backend/src/api/PostController.ts

```diff
@@ -260,13 +262,20 @@ export default class PostController {
 
     preview(request: APIRequest<PostPreviewRequest>, response: APIResponse<PostPreviewResponse>) {
         const userId = request.session.data.userId;
-        if (!userId) {
+        // allow access from private IPs with no session id
+        if (!userId && !isPrivate(request.ip)) {
```

Comment:
Please be extra careful here. Verify that IP comes from the actual user/CF and NOT from caddy. 

I'd suggest to add a special authorization token specifically for this case. 

Alternatively, as `media` container has DB access, you can create/use an actual session ID.